src/BuildAfterTargetingPack/BuildAfterTargetingPack.csproj

```diff
@@ -41,7 +41,7 @@
   <!-- Cannot build in source-build because that does not create an App.Ref layout. -->
   <Target Name="BuildDelayedProjects"
       BeforeTargets="Build"
-      Condition=" '$(DotNetBuildFromSource)' != 'true' "
+      Condition=" '$(DotNetBuild)' != 'true' "
```

Comment:
This part is what I think is using that local laid out SDK. Based on the project list, we need to remove this condition (ideally in source-only modes too).

Reply:
Here's what's installing the custom ref pack (the linked-to SDK issue even describes the same mechanism I mentioned in a different comment about how to work around this):

https://github.com/dotnet/aspnetcore/blob/5c17b92560953e86f1ea8502a567bcb3a1be4a2f/src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj#L255-L264